src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -286,12 +292,20 @@ public function __construct(array $attributes = [])
     protected function bootIfNotBooted()
     {
         if (! isset(static::$booted[static::class])) {
-            static::$booted[static::class] = true;
+            if (isset(static::$booting[static::class])) {
+                throw new Exception('"'.__METHOD__.'" cannot be called on the "'.static::class.'" class while it is already being booted.');
```

Comment:
Shouldn't it be a `LogicException`?

>  Exception that represents error in the program logic. This kind of exception should lead directly to a fix in your code. 

https://www.php.net/manual/en/class.logicexception.php